### PR TITLE
Prostate 3D data Segmentation with Improved 3D-UNet models

### PR DESCRIPTION
**Pull request by 4mark4444**
name: Zheng Yao
student number: s47697106

### CODE REVIEW

**Comment by celinsze** [comment]
**This is an initial inspection, no action is required at this point**

Recognition problem: total 14.5 Marks

Problem Solving: (4.5)
•	Unclear of where the evaluation results are obtained from: document mentioned the results for the average Dice score from the entire dataset instead of the test set but from the code, evaluation was performed on the test set only. 
•	Algorithm solves the problem and achieves a Dice score of above 0.7 for all labels assuming the results was calculated from the testing set as mentioned in train.py. 
Algorithm implementation: (3)
•	Algorithm implemented in Tensorflow for the UNet3D model and functions.
Design: Good coding design (1)
Commenting: Clear comments throughout the code. (1)
Difficulty: (5)
•	3D UNet – Normal

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Model operation is ok but reported partially.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. UNet3D (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is missing. </td><td>-2</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-10</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Segmentations not one hot encoded -2 additionally. Cant merge because of conflicting changes to main repo files (README). Please update for merge, doesn't affect grade.
